lib/collection/url.js

```diff
@@ -34,7 +34,7 @@ var _ = require('../util').lodash,
         extractPort: /:([^:]*)$/,
         extractPath: /.*?(?=\?|#|$)/,
         trimPath: /^\/((.+))$/,
-        extractQuery: /^\?([^#]+)/,
+        extractQuery: /^\?((([^#]*[=&%()==@[\]?:/.!-]))([^#{]*)|(\{\{([^]*?)}}*))*/g,
```

Comment:
This might break lots of things 😕
e.g `?q={{` is not matching.

Reply:
Isn't that expected behaviour ? If you don't have a closing }} it's not a valid syntax hence variable won't be resolved ?

Reply:
No, basic rule: anything in between `?` & `#` is the query string.
Allowing `#` in the variable name(inside query string) is what we have to support.

Reply:
As per rfc 3986, If data for a URI component would conflict with a reserved character's purpose as a delimiter, then the conflicting data must be percent-encoded before the URI is formed.

And it defines following as reserved characters

reserved = gen-delims / sub-delims

gen-delims = ":" / "/" / "?" / "#" / "[" / "]" / "@"

sub-delims = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="